packages/mgt-chat/src/components/ChatList/ChatList.tsx

```diff
@@ -52,47 +56,64 @@ export const ChatList = (
     IChatListItemInteractionProps &
     IChatListMenuItemsProps & {
       buttonItems?: ChatListButtonItem[];
+      chatThreadsPerPage: number;
     }
 ) => {
-  const { value } = props.dataContext as { value: GraphChat[] };
-  const chats: GraphChat[] = value;
-
   const styles = useStyles();
-  const chatClient: StatefulGraphChatListClient = useGraphChatListClient();
-  const [chatState, setChatState] = useState(chatClient.getState());
-  const [chatThreads, setChatThreads] = useState<GraphChat[]>(chats);
+  const [chatListClient, setChatListClient] = useState<StatefulGraphChatListClient | undefined>();
+  const [chatListState, setChatListState] = useState<GraphChatListClient | undefined>();
   const [menuItems, setMenuItems] = useState<ChatListMenuItem[]>(props.menuItems === undefined ? [] : props.menuItems);
-
-  const onChatListEvent = (state: ChatListEvent) => {
-    // TODO: implementation will happen later, right now, we just need to make sure messages are coming thru in console logs.
-
-    console.log(state.type);
-    console.log(state.message);
-  };
-
   const [selectedItem, setSelectedItem] = useState<string>();
 
+  // wait for provider to be ready before setting client and state
   useEffect(() => {
-    chatClient.onChatListEvent(onChatListEvent);
-    chatClient.onStateChange(setChatState);
-    return () => {
-      chatClient.offChatListEvent(onChatListEvent);
-      chatClient.offStateChange(setChatState);
-    };
-  }, [chatClient]);
-
-  const chatListButtonItems = props.buttonItems === undefined ? [] : props.buttonItems;
+    const provider = Providers.globalProvider;
+    provider.onStateChanged(evt => {
+      if (evt.detail === ProviderState.SignedIn) {
+        const client = new StatefulGraphChatListClient(props.chatThreadsPerPage);
+        setChatListClient(client);
+        setChatListState(client.getState());
+      }
+    });
 
-  useEffect(() => {
     const markAllAsRead = {
       displayText: 'Mark all as read',
-      onClick: () => {}
+      onClick: () => {
+        console.log('mark all as read');
+      }
     };
 
     menuItems.unshift(markAllAsRead);
     setMenuItems(menuItems);
   }, []);
 
+  useEffect(() => {
+    // handles events emitted from the chat list client
+    const handleChatListEvent = (event: ChatListEvent) => {
+      if (event.type === 'chatMessageReceived') {
+        if (props.onMessageReceived) {
+          props.onMessageReceived();
+        }
+      }
+    };
+    if (chatListClient) {
+      chatListClient.onStateChange(setChatListState);
+      chatListClient.onChatListEvent(handleChatListEvent);
```

Comment:
Is this old? I thought @seekdavidlee removed onChatListEvent()

Reply:
I added it back to handle this case, I'm guessing there's another way to accomplish this?